tests/v5/mqtt5_to_mqtt3_adapter_tests.c

```diff
@@ -2686,6 +2686,91 @@ static int s_mqtt5to3_adapter_subscribe_single_success_fn(struct aws_allocator *
 
 AWS_TEST_CASE(mqtt5to3_adapter_subscribe_single_success, s_mqtt5to3_adapter_subscribe_single_success_fn)
 
+/*
+ * This function tests receiving a subscribe acknowlege after disconnecting from
+ * the server.
+ * it expects a AWS_MQTT_QOS_FAILURE return
+ */
+static int s_mqtt5to3_adapter_subscribe_single_success_null_suback_fn(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+
+    aws_mqtt_library_init(allocator);
+
+    struct mqtt5_client_test_options test_options;
+    aws_mqtt5_client_test_init_default_options(&test_options);
+
+    test_options.server_function_table.packet_handlers[AWS_MQTT5_PT_SUBSCRIBE] =
+        s_mqtt5_mock_server_handle_subscribe_suback_success;
```

Comment:
Should we set this to NULL so that we guarantee that the server never respond with a `suback`?